modules/nf-core/ncbidatasets/download/main.nf

```diff
@@ -0,0 +1,108 @@
+process NCBIDATASETS_DOWNLOAD {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "conda-forge::ncbi-datasets-cli=15.11.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ncbi-datasets-pylib:15.11.0--pyhdfd78af_0':
+        'staphb/ncbi-datasets:15.11.0' }"
+
+    input:
+
+    val(meta)
+
+    output:
+    // Required output
+    tuple val(meta), path("*.zip")   ,             emit: zip
+    path "versions.yml"              ,             emit: versions
+    // Optional outputs, depends on tool usage
+    tuple val(meta), path("cds.fna.gz"),           emit: cds,        optional: true
+    tuple val(meta), path("gene.fna.gz"),          emit: gene,       optional: true
+    tuple val(meta), path("*genomic.fna.gz"),      emit: genome,     optional: true
+    tuple val(meta), path("genomic.gbff.gz"),      emit: gbff,       optional: true
+    tuple val(meta), path("genomic.gff.gz"),       emit: gff,        optional: true
+    tuple val(meta), path("genomic.gtf.gz"),       emit: gtf,        optional: true
+    tuple val(meta), path("protein.faa.gz"),       emit: protein,    optional: true
+    tuple val(meta), path("rna.fna.gz"),           emit: rna,        optional: true
+    tuple val(meta), path("3p_utr.fna.gz"),        emit: utr_3p,     optional: true
+    tuple val(meta), path("5p_utr.fna.gz"),        emit: utr_5p,     optional: true
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def valid_commands = ["genome accession", "genome taxon", "gene accession", "gene gene-id", "gene symbol", "gene taxon", "virus genome taxon", "virus genome accession", "virus protein"]
+    if (!valid_commands.contains(meta.command)) {
+        error "Unsupported command: ${meta.command} "
+    }
+
+
+    def args = task.ext.args ?: ''
+    def args_from_csv = meta.extra_args ?: ''
```

Comment:
same here. is this a csv file? In that case we need to add it as 'input' so the file can be properly staged :)

Reply:
This needs some explaining and I think that the naming is a bit off. I do use it now as coming from a nf-validation validated csv file through .fromSamplesheet(). But I think I will need some feedback on how to implement a few things while conforming to the nf-core standards. I will seek further advice on the nf-core modules channel on Slack

Reply:
ok sounds good. I think what you probably want is `tuple val(meta), path(csv)` and the csv is your samplesheet.